game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -0,0 +1,242 @@
+package games.strategy.triplea.delegate.battle.steps;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleStepStrings;
+import games.strategy.triplea.delegate.battle.MustFightBattle.ReturnFire;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Get the steps that will occurr in the battle */
+@Builder
+public class BattleSteps implements BattleStepStrings {
+
+  final @NonNull Boolean canFireOffensiveAa;
+  final @NonNull Boolean canFireDefendingAa;
+  final @NonNull Boolean showFirstRun;
+  final @NonNull GamePlayer attacker;
+  final @NonNull GamePlayer defender;
+  final @NonNull Collection<Unit> offensiveAa;
+  final @NonNull Collection<Unit> defendingAa;
+  final @NonNull Collection<Unit> attackingUnits;
+  final @NonNull Collection<Unit> defendingUnits;
+  final @NonNull Collection<Unit> attackingWaitingToDie;
+  final @NonNull Collection<Unit> defendingWaitingToDie;
+  final @NonNull Territory battleSite;
+  final @NonNull GameData gameData;
+  final @NonNull Collection<Unit> bombardingUnits;
+  final @NonNull Function<Collection<Unit>, Collection<Unit>> getDependentUnits;
+  final @NonNull Boolean isBattleSiteWater;
+  final @NonNull Boolean isAmphibious;
+  final @NonNull Supplier<Collection<Territory>> getAttackerRetreatTerritories;
+  final @NonNull BiFunction<GamePlayer, Collection<Unit>, Collection<Territory>>
+      getEmptyOrFriendlySeaNeighbors;
+
+  public List<String> get() {
```

Comment:
`get()` is the method for a `Supplier`, should this class implement  `Supplier<List<String>>`?

Otherwise, seeing the usage, I would have thought that this class were a supplier.

Reply:
No, this isn't supposed to be a supplier.  I'm just bad with names.